airflow-core/src/airflow/dag_processing/processor.py

```diff
@@ -409,6 +409,10 @@ def _handle_request(self, msg: ToManager, log: FilteringBoundLogger, req_id: int
         elif isinstance(msg, GetVariable):
             var = self.client.variables.get(msg.key)
             if isinstance(var, VariableResponse):
+                if var.value:
+                    from airflow.sdk.log import mask_secret
+
+                    mask_secret(var.value, var.key)
```

Comment:
Why here? We already have it in `task-sdk/src/airflow/sdk/execution_time/context.py:_get_variable`

Reply:
Good catch, yeah it is already handled for dag processor and works without needing this, however, it doesnt work as expected for triggerer if I remove it. I have left a TODO for it, and that can be handled as an immediate follow-up. Or if you know about it we could collab! I tried debugging a bit and couldnt get to the bottom of it

Reply:
>however, it doesnt work as expected for triggerer if I remove it.

Is there a test that fails, what's the error?